sparkly/testing.py

```diff
@@ -214,65 +235,234 @@ def assertEqual(self,
+                        for v in sequence
+                    ])
+                )
+
+            def __lt__(self, other):
+                # None is not a nice value to compare to when trying to
+                # order things, as TypeErrors are raised. Instead, we
+                # transform it to a tuple - if the first entry doesn't
+                # match (is it None?) we don't need to compare further
+                def _neutralize_none(entry):
+                    if isinstance(entry, tuple):
+                        return entry is None, tuple(_neutralize_none(e) for e in entry)
+                    return entry is None, entry
+                return [_neutralize_none(e) for e in self] < [_neutralize_none(e) for e in other]
+
+
+        # Define new equality asserters for floats, rows and datatypes
+        def assert_float_equal(self, float1, float2, msg=None):
+            if not isinstance(float1, Float):
+                float1 = Float(float1)
+            if not isinstance(float2, Float):
+                float2 = Float(float2)
+
+            if float1 == float2:
+                return
+
+            if not atol and not rtol:
+                standard_msg = '{} != {}'.format(float1, float2)
+            else:
+                standard_msg = (
+                    '{} != {} within absolute tolerance {} and relative tolerance {}'
+                    .format(float1, float2, atol, rtol)
+                )
+            self.fail(self._formatMessage(msg, standard_msg))
+
+        def assert_row_equal(self, row1, row2, msg=None):
+            self.assertEqual(Row(row1), Row(row2), msg)
+
+        def assert_datatype_equal(self, dt1, dt2, msg=None):
+            if not isinstance(dt1, DataType):
+                dt1 = DataType(dt1)
+            if not isinstance(dt2, DataType):
+                dt2 = DataType(dt2)
+
+            if dt1 != dt2:
+                standard_msg = '{} != {}'.format(safe_repr(dt1, True), safe_repr(dt2, True))
+                diff = '\n' + '\n'.join(difflib.ndiff(dt1.pretty_repr(), dt2.pretty_repr()))
+                standard_msg = self._truncateMessage(standard_msg, diff)
+                self.fail(self._formatMessage(msg, standard_msg))
+
+        # Create a context manager to temporarily register our asserters,
+        # then restore them to defaults after this function is finished
+        # since they might depend on specific parameters provided here
+        # (e.g., atol/rtol for floats)
+        @contextlib.contextmanager
+        def temp_add_type_equality_func(self, typeobj, function):
+            old_asserter = self._type_equality_funcs.get(typeobj)
+            self.addTypeEqualityFunc(typeobj, function)
+            yield
+            self.addTypeEqualityFunc(typeobj, old_asserter)
+
+        temp_add_type_equality_func = partial(temp_add_type_equality_func, self)
+
+        # Register equality asserters
+        with temp_add_type_equality_func(float, partial(assert_float_equal, self)), \
```

Comment:
i actually think it's more cumbersome to have to pass an "ignore_order" and "ignore_nullability" and float precision arguments to all of your assert_equals calls, instead of just setting it once like we do with maxDiff in a lot of places.  I agree that it would be cumbersome if you expect to call assertEqual with different values for those paramters pretty often in the same test, but I think that will be the exceptional case.  

I think those params make sense as part of an assertEqual call when the assert equal call is specific to a single type of input, like assertAlmostEqual() or assertListEqual(), but now you are adding 5 different params that govern 3 different types of functionality into one interface which makes it much more confusing.  The param names like atol and rtol are only NOT confusing in cases like numpy.testing because you are calling them in functions where the whole purpose is to do comparison with tolerances.  

If you really want to make it easy to mix your assertEqual with vanilla assertEqual you shouldn't be overriding it; just name this function something different.  Unregistering isn't really necessary because registering only registers for this instance of the unittest.testcase.  Other testcases can still use vanilla assertEqual.